File: programers/javascript/K번째수.js
array = [1, 5, 2, 6, 3, 7, 4];
commands = [
  [2, 5, 3],
  [4, 4, 1],
  [1, 7, 3],
  [3, 4, 1],
];

console.log(array.slice(2 - 1, 5).sort()[3 - 1]);
// return = [5,6,3]
function solution(array, commands) {
  var answer = [];
  for (let a = 0; a < commands.length; a++) {
    var i = commands[a][0];
    var j = commands[a][1];
    var k = commands[a][2];

    const value = array.slice(i - 1, j).sort((a, b) => {
      return a - b;
    });
    answer.push(value[k - 1]);
  }
  return answer;
}

console.log(solution(array, commands));

File: programmers/K번째수.py
# 정렬
# level 1

array = [1, 5, 2, 6, 3, 7, 4]
commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]


def solution(array, commands):
    answer = []
    for command in commands:
        i = command[0]
        j = command[1]
        sorted_array = sorted(array[i - 1:j])
        answer.append(sorted_array[command[2] - 1])

    return answer


print(solution(array, commands))
